### PR TITLE
fix(urlSync): urls should be safe by default

### LINKED ISSUE #982: Urls should be safe by default
Following https://github.com/algolia/instantsearch.js/issues/848#issuecomment-191795411 and a support request we had also recently, I believe we should fix our urls.

We will get more and more issues about unsafe urls in the future.

## Problem :trollface: 

we implemented urls so that they do not look too bad:
`?query=&hits=24&idx=instant_search&p=0&nR[price][<=][0]=320&is_v=1`

But this is not really standard because most parameters should be encoded so that our urls are compatible with every use case (copy pasting from/to email/app/slack, server side parsing, library parsing..)

Thus it would mean having urls like:

`?query=&hits=24&idx=instant_search&p=0&nR%5Bprice%5D%5B%3C%3D%5D%5B0%5D=380&is_v=1`

Not so clean and readable but would work everywhere.

## Possible solutions :horse: 

- allow the helper to always encode. It's currently [disabled](https://github.com/algolia/algoliasearch-helper-js/blob/4357cf95d7efb0c1d017a0c897093289bf31c266/src/url.js#L157) and [we specifically asked](https://github.com/ljharb/qs/issues/116) for this feature in the `qs` project (lol)

This even seem to be backward compatible. It would be a WTF moment for developers but safety first!

- create a brand new stringifier for complex objects => url which would not be based on unsafe characters like [ ] > =

Here we should be careful about URL versions if we want to stay backward compatible (is_v= will help)

- use https://github.com/vjeux/URLON

Help and feedback appreciated.

### PR DESCRIPTION
**Pull request by vvo**
tl;dr; Urls stored in the browser bar were made to look not too bad by omitting to encode some characters. Turns out it was a bad idea, copy pasting the url from browser bar to slack or github failed.

Fixes #982 #848 by using the new safe: true option from the helper when asking for a state as a url (https://github.com/algolia/algoliasearch-helper-js/pull/317).

This seems to be backward compatible (instantsearch wise):
- for current implementations, urls in the browser bar will move from `&hRF[color]` to `&hRF%05Bcolor%05D`
- urls sent to the server (if it happen people were reading them) will move from `&hRF[color]` to `&hRF%05Bcolor%05D` but I believe most server side code is already using decodeURI
- no change to current urls, they will be properly decoded (no change was needed on the decoding part)

If you have any idea how this might break people's website, please add a comment.

cc @olance

### CODE REVIEW

**Comment by bobylito** [comment]
No config for this behaviour?

**Comment by vvo** [comment]
> No config for this behaviour?

I first started doing it but then thought "well, the current urls are broken, do we want to allow developers to have `broken` urls?"

**Comment by bobylito** [comment]
Lgtm :) :+1:
